File: 210/level1/로또의최고순위와최저순위.java
// 20220801

package level1;

import java.util.Arrays;

class Main {
    public static void main(String[] args) throws Exception {
        int[] lottos = {44, 1, 0, 0, 31, 25};
        int[] win_nums = {31, 10, 45, 1, 6, 19};

        Solution sol = new Solution();
        System.out.println("result : " + sol.solution(lottos, win_nums)[0] + " " + sol.solution(lottos, win_nums)[1]);
    }
}


class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {7, 7};
        for (int i = 0; i < 6; i++) {
            if (lottos[i] == 0) {
                answer[0]--;
            } else {
                int a = lottos[i];
                if (Arrays.stream(win_nums).anyMatch(j -> j == a)) {
                    answer[0]--;
                    answer[1]--;
                }
            }
        }
        answer[0] = answer[0] == 7 ? 6 : answer[0];
        answer[1] = answer[1] == 7 ? 6 : answer[1];

        return answer;
    }
}

File: Hprogram/level1/로또의최고순위와최저순위.js
function solution(lottos, win_nums) {
	const nice = [0, 0];
	// 몇개가 맞으면 등수가 몇등인지 테이블로 만들어줌
	const result = { 6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6 };

	// lottos의 데이터를 비교해 당첨 번호라면 양쪽에 모두 더하여주고
	// 0 이라면 최고 등수에만 더해줌
	for (let i = 0; i < lottos.length; i++) {
		if (lottos[i] === 0) {
			nice[0] = nice[0] + 1;
		} else if (win_nums.includes(lottos[i])) {
			nice[0] = nice[0] + 1;
			nice[1] = nice[1] + 1;
		}
	}

	// 미리 저장시켜둔 테이블을 이용해 해당 등수를 출력
	return [result[nice[0]], result[nice[1]]];
}
